src/classes/ADDR_Validator_TEST.cls

```diff
@@ -475,9 +419,487 @@ public with sharing class ADDR_Validator_TEST {
 
         ctrl.verifyAddressButton();
 
-        updatedAddress = [SELECT Id, MailingStreet__c, Verified__c, Verification_Status__c FROM Address__c WHERE Id = :av[0].Id];
+        updatedAddress = getAddress(av[0].Id);
         System.assertEquals(true, updatedAddress.Verified__c);
         System.assertEquals(Label.Addr_Verified, updatedAddress.Verification_Status__c, 'Address should have been verified');
         System.assertEquals('1 Infinite Loop', updatedAddress.MailingStreet__c);
     }
+
+    /*******************************************************************************************************
+    * @description Verify the page does not display API key and other address verification settings
+    * when user has no permission to view the page
+    */
+    @isTest
+    private static void shouldNotRetrieveApiSettingsWhenUserHasNoAccess() {        
+        createDefaultSettings();
+        
+        Test.setCurrentPage(Page.STG_PanelAddrVerification);
+        AddressVerifierExtension addrVerifier = new AddressVerifierExtension().returnHasAccess(false);
+        STG_PanelAddrVerification_CTRL.addrVerifier = addrVerifier;
+        STG_PanelAddrVerification_CTRL ctrl = new STG_PanelAddrVerification_CTRL();
+
+        UTIL_UnitTestData_TEST.assertPageHasMessage(System.Label.stgAddrPermissionDenied, ApexPages.Severity.INFO);
+        assertAuthIdAndTokenAreNull(ctrl);
+
+        ctrl.initDynamicFields();
+        assertAuthIdAndTokenAreNull(ctrl);
+
+        ctrl.apiTestEntry = ZIP_CODE;
+        ctrl.testApi();
+        System.assertEquals(System.Label.stgAddrPermissionDenied, ctrl.apiTestResult);
+        assertAuthIdAndTokenAreNull(ctrl);
+    }
+
+    /*******************************************************************************************************
+    * @description Verify permission denied message is displayed on health check panel
+    * when read only user verifies address
+    */
+    @isTest
+    private static void healthCheckShouldDisplayPermissionDeniedMessageWhenUserHasNoAccess() { 
+        Addr_Verification_Settings__c settings = createDefaultSettings();
```

Comment:
This var is never used. There is quite a bit of code in this class that assigns the result of `createDefaultSettings();` to a never used var as well.

Reply:
The var assignment has been removed.